src/context.rs

```diff
@@ -821,12 +821,26 @@ impl Context {
 
         let secret_key = &load_self_secret_key(self).await?.primary_key;
         let key_created = secret_key.created_at().timestamp();
-        res += &format!("key_created {}", key_created);
+        res += &format!("key_created {}\n", key_created);
+
+        let self_reporting_id = match self.get_config(Config::SelfReportingId).await? {
+            Some(id) => id,
+            None => {
+                let id = create_id();
+                self.set_config(Config::SelfReportingId, Some(&id)).await?;
```

Comment:
The function is called `get_self_report()`, but it also can change the config and actually can be called from parallel threads. I think it's better to generate and store this SelfReporingId earlier.

Reply:
OTOH, it's a private function called only from `draft_self_report`, and why would anyone call `draft_self_report` from multiple threads. But I can also move the creation into migrations.